### DISCUSSION TITLE
Load RDF from external url

### DISCUSSION THREAD

**Discussion by sandrarum** [eclipse-rdf4j/rdf4j#3294] [Q&A]
I use the Rest API to work with the RDF4J repositories.

I want to load RDF data from an external url directly into a repository.

I read the documentation and found several ways to load RDF data. It always works by setting the RDF data as content (e.g. https://rdf4j.org/documentation/reference/rest-api/#the-add-operation). This means I have to download the RDF from the source url beforehand (buffer the data locally) to add the RDF data as request data.

Is there a possibility to specify the external url directly when calling the Rest API in order to avoid buffering?

**Comment by abrokenjester** [comment]
Yes, this is possible, by using a SPARQL Update operation as the payload, rather than the RDF document itself. See https://rdf4j.org/documentation/reference/rest-api/#perform-update-described-in-a-sparql-11-update-string for an example. 

The update operation you want to use is `LOAD <http://uri-of-rdf-document>` - see https://www.w3.org/TR/sparql11-update/#load .

As an aside, if you want to avoid having to put the update in form-encoded (as in the documentation example), you can instead also do a POST request with Content-Type set to `application/sparql-update` and then have the SPARQL update string as plain text (unencoded) in the request body. So, like this:

```
POST /rdf4j-server/repositories/mem-rdf/statements 
Content-Type: application/sparql-update

LOAD <http://uri-of-rdf-document>
```
